### PR TITLE
Build with Section and NoTest flags

### PR DESCRIPTION
**Pull request by ikaneshiro**
This adds functionality for the `section` and `notest` flags by giving the build system the ability to selectively run the different sections of any given definition. This also includes changes to the CLI in order to use these flags. 

This also cleans up build output to be more readable and useful when performing a normal build. 

Attn: @singularityware-admin

### CODE REVIEW

File: src/pkg/build/build.go [hunk 1]
```diff
@@ -326,6 +339,24 @@ func makeDef(spec string) (types.Definition, error) {
 	return def, nil
 }
 
+func (b *Build) addOptions() {
+	b.b.NoTest = b.noTest
+	b.b.Sections = b.sections
+}
+
+// RunSection determines if a section name was specified
+func (b Build) runPre() bool {
```
Comment (ArangoGutierrez) [inline src/pkg/build/build.go]: Func name != comment name

File: src/pkg/build/types/bundle.go [hunk 2]
```diff
@@ -68,3 +70,16 @@ func NewBundle(directoryPrefix string) (b *Bundle, err error) {
 func (b *Bundle) Rootfs() string {
 	return filepath.Join(b.Path, b.FSObjects["rootfs"])
 }
+
+// RunSection determines if a section name was specified
```
Comment (ArangoGutierrez) [inline src/pkg/build/types/bundle.go]: was specified on the CLI, could you be more verbose on this comment.

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: src/runtime/engines/imgbuild/process.go [hunk 3]
```diff
@@ -87,3 +109,47 @@ func (e *EngineOperations) CleanupContainer() error {
 func (e *EngineOperations) PostStartProcess(pid int) error {
 	return nil
 }
+
+func (e *EngineOperations) insertEnvScript() error {
+	if e.EngineConfig.RunSection("environment") && e.EngineConfig.Recipe.ImageData.Environment != "" {
+		sylog.Infof("Adding environment to container")
+		err := ioutil.WriteFile("/.singularity.d/env/90-environment.sh", []byte("#!/bin/sh\n\n"+e.EngineConfig.Recipe.ImageData.Environment+"\n"), 0775)
```
Comment (jmstover) [inline src/runtime/engines/imgbuild/process.go:116]: @ikaneshiro 
As discussed, don't write out `91-environment.sh`. Just set `SINGULARITY_ENVIRONMENT` and let the usage of that in `%post`, etc... create the file.

Comment (ikaneshiro) [reply src/runtime/engines/imgbuild/process.go:116]: resolved by commit 5f01168

File: src/runtime/engines/imgbuild/process.go [hunk 4]
```diff
@@ -18,37 +19,58 @@ import (
...
+			sylog.Fatalf("failed to start %%post proc: %v\n", err)
+		}
+		if err := post.Wait(); err != nil {
+			sylog.Fatalf("post proc: %v\n", err)
+		}
 	}
-	if err := post.Wait(); err != nil {
-		sylog.Fatalf("post proc: %v\n", err)
+
+	//append environment
```
Comment (ArangoGutierrez) [inline src/runtime/engines/imgbuild/process.go]: // + space 
;)

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

**Review by ArangoGutierrez — APPROVED** [review]

**Comment by ArangoGutierrez** [comment]
@bauerm97  RFR

**Pull request merged**